src/universal/components/QueryRenderer/QueryRenderer.js

```diff
@@ -161,6 +178,27 @@ export default class ReactRelayQueryRenderer extends React.Component {
       this._selectionReference.dispose();
       this._selectionReference = null;
     }
+  };
+
+  _subscribe(subscriptions) {
+    if (subscriptions) {
+      const {environment, variables} = this._relayContext;
+      // subscribe to each new sub, or return the subKey of an already existing sub
+      const subscriptionKeys = subscriptions.map((sub) => sub(environment, variables));
+      // provide an unsub prop to the component so we can unsub whenever we want
+      // when we call unsub we want to:
+      //   release immediately if component is unmounted
+      //   set releaseOnUnmount to true if component is still mounted
+      //
+      this.unsubscribe = () => {
+        if (this._mounted) {
```

Comment:
I bet all this `_mounted` business was a _hell_ of a lot of work to test and get right.